server/chat/backend/agent/tools/bitbucket/repos_tool.py

```
@@ -38,10 +46,108 @@ class BitbucketReposArgs(BaseModel):
+        pattern = re.compile(query)
+    except re.error:
+        return formatted, total  # not a regex either — genuinely no matches
+
+    # A daemon thread (NOT concurrent.futures: its non-daemon workers are
+    # joined at interpreter exit, so an abandoned catastrophic scan would
+    # block process shutdown). On timeout the runaway thread is abandoned —
+    # it burns CPU until the scan finishes but never blocks exit, and it
+    # holds only this call's line list.
+    outcome: dict = {}
+
+    def _regex_scan():
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `_regex_scan`

Add return type annotation: `None`

(ANN202)